example/ZXingQtReader.h

```diff
@@ -497,14 +511,24 @@ inline void registerQmlAndMetaTypes()
 
 	// supposedly the Q_DECLARE_METATYPE should be used with the overload without a custom name
 	// but then the qml side complains about "unregistered type"
-	qRegisterMetaType<ZXingQt::Position>("Position");
-	qRegisterMetaType<ZXingQt::Barcode>("Barcode");
+	qmlRegisterType<ZXingQt::Position>("ZXingQt", 1, 0, "position");
+	qmlRegisterType<ZXingQt::Barcode>("ZXingQt", 1, 0, "barcode");
```

Comment:
Why switch to `qmlRegisterType` here and why use a lower case type name (e.g. `position`)?

Reply:
I'm not sure why you used `qRegisterMetaType` in the first place, as based on the comment it seems that your intention was to expose it to QML.

Exposing types to QML should be done using `qmlRegisterType`. This allow the type to be used for a QML property, i.e.

    // QML file
    ...
    property barcode myBarcode: ...
    ...

Value types (i.e. classes using the `Q_GADGET` macro instead of the `Q_OBJECT` macro), should be exposed to QML using a lower case name according to the [docs](https://doc.qt.io/qt-6/qtqml-cppintegration-definetypes.html#value-types-with-enumerations):

> In contrast to object types, value types require lower case names.

Reply:
> I'm not sure why you used `qRegisterMetaType` in the first place

I don't remember either. It was most likely derived from some sample/tutorial code. If `QML_VALUE_TYPE` is the way to go for QT6, then I'd prefer that. Given the imminent deprecation of Qt5, I'd prefer to completely drop QT5 support before 'releasing' this as an 'official' wrapper anyway.

Regarding the lower-case naming, I'll need to read some docs to get the reasoning and implication, it seems.

Reply:
I tried using `QML_VALUE_TYPE`, but that also requires changes to the CMake project (see https://doc.qt.io/qt-6/qqmlintegration-h-qtqml-proxy.html#QML_ELEMENT), which seems undesirable to me if you want to expose it as a header only wrapper. So, I would propose to keep using `qmlRegisterType` for now (this function is not deprecated by Qt as far as I know).

Reply:
I agree. Thanks for looking into it.